bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -91,16 +93,18 @@ private void loadConfig() {
         Object type = module.getConfiguration().get(SCRIPT_TYPE);
         Object script = module.getConfiguration().get(SCRIPT);
         if (!isValid(type)) {
-            throw new IllegalStateException(String.format("Type is missing in the configuration of module '%s'.", module.getId()));
+            throw new IllegalStateException(
```

Comment:
You could add a method `String resolveValidType(final Object type)` that throws a ISE if the type is not valid and otherwise return a valid type.

Reply:
Thanks for the idea. I introduces a new method to get a valid config and assign it to the properties inside the contructor. This solution solves your other https://github.com/openhab/openhab-core/pull/910#discussion_r300031597 too and made it possible to make the properties final.